hidimstat/cpi.py

```diff
@@ -0,0 +1,227 @@
+import numpy as np
+from joblib import Parallel, delayed
+from sklearn.base import (
+    BaseEstimator,
+    TransformerMixin,
+    check_is_fitted,
+    clone,
+)
+from sklearn.metrics import mean_squared_error
+
+
+class CPI(BaseEstimator, TransformerMixin):
+    """
+    Conditional Permutation Importance (CPI) algorithm.
+    :footcite:t:`Chamma_NeurIPS2023` and for group-level see
+    :footcite:t:`Chamma_AAAI2024`.
+
+
+    Parameters
+    ----------
+    estimator: scikit-learn compatible estimator
+        The predictive model.
+    imputation_model: scikit-learn compatible estimator or list of
+    estimators
+        The model(s) used to estimate the covariates. If a single estimator is
+        provided, it will be cloned for each covariate. Otherwise, a list of
+        potentially different estimators can be provided, the length of the
+        list must match the number of covariates.
+    n_perm: int, default=50
+        Number of permutations to perform.
+    groups: dict, default=None
```

Comment:
I think that groups are a data-dependent thing and should be provided at fit time ?

Reply:
Good point I will then move the `self._list_imputation_models` to the `.fit` function. 
Maybe it could also support clustering methods rather than predefined groups (for a next PR).